src/elements/SwitchToggle.vue

```diff
@@ -1,29 +1,46 @@
 <script setup lang="ts">
-import { ref, onMounted } from 'vue';
+import { ref, onMounted, computed } from 'vue';
 import { t } from '@/composable/i18n';
 
+const model = defineModel<boolean>();
+
 // component emits
 const emit = defineEmits(['changed']);
 
 // component properties
 interface Props {
   name: string;
-  active: boolean; // initial toggle state
+  active?: boolean; // deprecated in favor of v-model, sets initial toggle state
   disabled?: boolean; // flag for making toggle non changable
   label?: string; // input label
   noLegend?: boolean; // hide "on" and "off" labels
 }
 const props = defineProps<Props>();
 
-// current state
-const state = ref(false);
+// Internal state ref, used only when v-model is not provided
+const internalState = ref(false);
+
 onMounted(() => {
-  state.value = props.active;
+  // Initialize internal state only if v-model is not being used
+  if (model.value === undefined) {
+    internalState.value = props.active;
+  }
 });
+
+const currentState = computed(() => (model.value !== undefined ? model.value : internalState.value));
```

Comment:
`currentState` reflects correct value, abstracts away whether we are managing state internally or reactively